src/main/java/mods/eln/misc/DirectionSet.kt

```diff
@@ -0,0 +1,46 @@
+package mods.eln.misc
+
+import net.minecraft.nbt.NBTTagCompound
+import java.io.DataInputStream
+import java.io.DataOutputStream
+import java.util.*
+
+class DirectionSet: TreeSet<Direction>() {
+    fun toInt(): Int {
+        var value = 0
+        if (contains(Direction.XN)) value += 1
```

Comment:
I came up with something like this:

fun Int.setBit(position: Int): Int = this or (1 shl position)
fun Int.isBitSet(position: Int): Boolean = this and (1 shl position) != 0

Do you know if it would be possible to change the value of "this" in an extension function?